lightkurve/targetpixelfile.py

```diff
@@ -375,10 +380,60 @@ def _parse_aperture_mask(self, aperture_mask):
                 aperture_mask = np.ones((self.shape[1], self.shape[2]), dtype=bool)
             elif aperture_mask == 'pipeline':
                 aperture_mask = self.pipeline_mask
+            elif aperture_mask == 'threshold':
+                if 'threshold' in kwargs:
+                    threshold = kwargs['threshold']
+                else:
+                    threshold = 3.
+                aperture_mask = self._threshold_mask(threshold)
         self._last_aperture_mask = aperture_mask
         return aperture_mask
 
-    def centroids(self, aperture_mask='pipeline'):
+    def _threshold_mask(self, threshold):
+        """Identify region that is above the threshold and contiguous
+        with the central pixel
+
+        Parameters
+        ----------
+        threshold : float
+            A value for the number of sigma above the background to draw the
+            aperture.
+
+        Returns
+        -------
+        aperture_mask : ndarray
+            2D boolean numpy array containing `True` for selected pixels.
+        """
+
+        # calculate a median image
+        median_image = np.nanmedian(self.flux + self.flux_bkg, axis=0)
+        vals = median_image[np.isfinite(median_image)].flatten()
+
+        # calculate what the theshold value is in counts
+        mad_cut = (1.4826 * MAD(vals) * threshold) + np.median(median_image)
+
+        # identify pixel above the threshold and label contiguous regions
+        threshold_region = np.where(median_image > mad_cut, 1, 0)
+        labeled_values = label(threshold_region)[0]
+
+        # find the central pixel
+        image_shape = np.shape(median_image)
+        central_pixel = [1 + image_shape[0] // 2, 1 + image_shape[1] // 2]
+
+        #find brightest pix within 5x5 of central pix
+        #this assumes target is close to center of postage stamp
+        median_image_center = median_image[central_pixel[0] - 2:central_pixel[0] + 2,
+                                           central_pixel[1] - 2:central_pixel[1] + 2]
+        brightest_pixel = np.unravel_index(median_image_center.argmax(),
+                                        median_image_center.shape)
+        brightest_pixel_y, brightest_pixel_x = brightest_pixel
+
+        region_value = labeled_values[central_pixel[0] + brightest_pixel_y - 2,
```

Comment:
Why are we using 5x5 pixels here? Should this contiguous region size be an additional parameter? What if I'm looking at a galaxy, a saturated star or at a faint source in a crowded region?